core/src/test/scala/unit/kafka/server/ServerStartupTest.scala

```diff
@@ -44,6 +43,47 @@ class ServerStartupTest extends ZooKeeperTestHarness {
   }
 
   @Test
+  def testConflictBrokerStartupWithSamePort {
+    // Create and start first broker
+    val brokerId1 = 0
+    val zkChroot1 = "/kafka-chroot-for-unittest"
+    val props = TestUtils.createBrokerConfig(brokerId1, zkConnect)
+    val zooKeeperConnect = props.get("zookeeper.connect")
+    props.put("zookeeper.connect", zooKeeperConnect + zkChroot1)
+    val server1 = TestUtils.createServer(KafkaConfig.fromProps(props))
+
+    // Retrieve the broker port for server1
+    val brokerZkPath = zkChroot1 + ZkUtils.BrokerIdsPath + "/" + brokerId1
+    val pathExists = zkUtils.pathExists(brokerZkPath)
+    assertTrue(pathExists)
+
+    val port = zkUtils.readDataMaybeNull(brokerZkPath)._1 match {
```

Comment:
You can get the port from `server1` instead of reading it from ZK. Also, why are we using a different zk chroot?